src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java

```diff
@@ -71,7 +76,16 @@ public void postLoanPolicyStorageLoanPolicies(
     Context vertxContext) {
 
     PgUtil.post(LOAN_POLICY_TABLE, entity, okapiHeaders, vertxContext,
-        PostLoanPolicyStorageLoanPoliciesResponse.class, asyncResultHandler);
+        PostLoanPolicyStorageLoanPoliciesResponse.class, reply -> {
+      if (isInvalidUUIDError(reply)) {
```

Comment:
We should add UUID pattern to the `loan-policy.json` instead:
`"$ref": "raml-util/schemas/uuid.schema"`

Reply:
Hmm... it is going to be a breaking change then... Is this logic has introduced to fix `org.folio.rest.api.LoanPoliciesApiTest#loanPolicyWithFixedDate`?

Reply:
The logic was indeed added to fix the tests in the same way that kind of logic was already being used elswhere (such as `isSamePositionInQueueError()` from `RequestsAPI.java`).

Reply:
Ok, I think we can do the following then:
* Fix `loanPolicyWithFixedDate` test by providing a random UUID instead of the invalid `1234567890` string;
* Raise a separate jira to add UUID pattern for the `loan.fixedDueDateScheduleId` property (it will be considered as a breaking change).

@marcjohnson-kint what do you think?

Reply:
Hmm, am I misunderstanding this part of the change.

An invalid UUID should be different to a unique constraint error, are they being reported as the same?

Reply:
@bohdan-suprun Agreed, adding the UUID regular expression constraint to the interface will be a breaking change.

What does this change currently do, try to detect invalid UUIDs or not?

Reply:
The test in question is testing to confirm that an invalid uuid throws an invalid uuid (`LoanPoliciesApiTest.loanPolicyWithFixedDate`). This appears to be a test checking for a bad foreign key due to an invalid UUID being passed.

Its expecting this message: `Bad foreign key`, with a 422 response.

Instead, it gets a 500 response with `invalid input syntax for type uuid: "1234567890"` as the message.

To me like an invalid UUID is a bad foreign key. If the test is intended to not test that an invalid UUID could be a bad foreign key but instead to simply test for a non-existent UUID, then I believe the suggestion from @bohdan-suprun makes the most sense.

Reply:
> What does this change currently do, try to detect invalid UUIDs or not?

The change currently tries to detect invalid UUIDs.

Reply:
@kaladay What caused this behaviour to change during the upgrade?

Reply:
@marcjohnson-kint  I didn't investigate into RMB. I only isolated the issue specifically to the upgrade from `26.2.3` into `27.1.0`, with a minimal set of required changes (removing `fullText` column from `schema.json` and replacing `HttpStatus.HTTP_VALIDATION_ERROR.toInt())` with `422`).

Glancing at the diff in RMB between tags `v26.2.3` and `v27.1.0`, specifically for `domain-models-runtime`, the following seem likely places to start investigating:
- There is a brand new class `PgExceptionFacade.java`.
- The PgUtil also adds `422` responses as well as new validation pattern matchers.

Investigating the cause for that behavioral change in RMB seems to me to be out of the scope of this issue.

Reply:
> @marcjohnson-kint I didn't investigate into RMB. I only isolated the issue specifically to the upgrade from `26.2.3` into `27.1.0`, with a minimal set of required changes (removing `fullText` column from `schema.json` and replacing `HttpStatus.HTTP_VALIDATION_ERROR.toInt())` with `422`).

Hmm, are they not the same?

> Glancing at the diff in RMB between tags `v26.2.3` and `v27.1.0`, specifically for `domain-models-runtime`, the following seem likely places to start investigating:
> 
> * There is a brand new class `PgExceptionFacade.java`.
> * The PgUtil also adds `422` responses as well as new validation pattern matchers.
> 
> Investigating the cause for that behavioral change in RMB seems to me to be out of the scope of this issue.

Agreed, I think the minimum amount of changes should be made to perform the upgrade and preserve as much of the existing behaviour as possible.

I think we should consider changes to messages, not to be part of the preservation, and adapt the tests to reflect the messages produced (as long as they can be done consistently and we think they are triggered for the right reasons).

And then raise separate technical debt issues for improving some of these areas in future. Which could include investigating RAML Module Builder more.